hazelcast/src/test/java/com/hazelcast/query/impl/TypeConverterTest.java

```diff
@@ -195,9 +197,9 @@ public void testBigDecimalConvert_whenPassedStringValue_thenConvertToBigDecimal(
     }
 
     @Test
-    public void testBigDecimalConvert_whenPassedDoubleValue_thenConvertToBigDecimal() throws Exception {
+    public void testBigDecimalConvert_whenPassedDoubleValue_thenConvertToBigDecimal() {
         Double doubleValue = 3.141593;
```

Comment:
I think we need to tweak this test case a bit (with a comment) to fail with the constructor based approach and pass with `valueOf` to guarantee that we never return to the constructor based one. 

Same for the float case below.

Reply:
Do you mean to add a second failing test case for float + double?

Reply:
I've enhanced the tests to check for both values
```java
        Double doubleValue = 3.141593;
        Comparable expectedDecimal = BigDecimal.valueOf(doubleValue);
        Comparable unexpectedDecimal = new BigDecimal(doubleValue);

        Comparable comparable = TypeConverters.BIG_DECIMAL_CONVERTER.convert(doubleValue);

        assertThat(comparable, allOf(
                is(instanceOf(BigDecimal.class)),
                is(equalTo(expectedDecimal)),
                not(equalTo(unexpectedDecimal))
        ));
```

Reply:
:thumbsup: